components/apimgt/org.wso2.carbon.apimgt.common.gateway/src/main/java/org/wso2/carbon/apimgt/common/gateway/graphql/GraphQLOperationAnalyzer.java

```diff
@@ -0,0 +1,556 @@
+            if (variableMap.containsKey(valueFromFirstMap)) {
+                Object valueFromSecondMap = variableMap.get(valueFromFirstMap);
+                if (valueFromSecondMap instanceof Map) {
+                    Map<?, ?> innerMap = (Map<?, ?>) valueFromSecondMap;
+                    String[] keysArray = innerMap.keySet().toArray(new String[0]);
+                    combinedMap.put(keyFromFirstMap, keysArray);
+                } else {
+                    combinedMap.put(keyFromFirstMap, valueFromSecondMap);
+                }
+            }
+        }
+        return combinedMap;
+    }
+
+    private static Map<String, Object> transformMaps(Map<String, String> definedTypes,
+                                                     Map<String, Object> mutatedTypes) {
+        Map<String, Object> resultMap = new HashMap<>();
+        for (Map.Entry<String, String> entry : definedTypes.entrySet()) {
+            String newKey = entry.getValue();
+            String oldKey = entry.getKey();
+            Object newValue = mutatedTypes.get(oldKey);
+            resultMap.put(newKey, newValue);
+        }
+        return resultMap;
+    }
+
+    private static List<Map<String, Object>> convertMapToListOfMaps(Map<String, Object> inputMap) {
+        List<Map<String, Object>> result = new ArrayList<>();
+        for (Map.Entry<String, Object> entry : inputMap.entrySet()) {
+            Map<String, Object> newMap = new HashMap<>();
+            newMap.put("type", entry.getKey());
+            newMap.put("field", entry.getValue());
+            result.add(newMap);
+        }
+        return result;
+    }
+
+    private static Map<String, Object> getMutatedTypesFromOperation(List<Argument> arguments) {
+        Map<String, Object> mutatedTypes = new HashMap<>();
+        for (Argument argument : arguments) {
+            Object value = null;
+            if (argument.getValue() instanceof ObjectValue) {
+                List<String> fields = new ArrayList<>();
+                for (ObjectField field : ((ObjectValue) argument.getValue()).getObjectFields()) {
+                    value = field.getName();
+                    fields.add((String) value);
+                }
+                mutatedTypes.put(argument.getName(), fields);
+            } else if (argument.getValue() instanceof EnumValue) {
+                mutatedTypes.put(argument.getName(), ((EnumValue) argument.getValue()).getName());
+            } else if (argument.getValue() instanceof VariableReference) {
+                mutatedTypes.put(argument.getName(), ((VariableReference) argument.getValue()).getName());
+            }
+        }
+        return mutatedTypes;
+    }
+
+    /**
+     * @param arguments
+     * @return
+     */
+    public static Map<String, String> getDefinedTypesFromSchema(List<GraphQLArgument> arguments) {
+        Map<String, String> definedTypes = new HashMap<>();
+        for (GraphQLArgument argument : arguments) {
+            String type = null;
```

Comment:
Let's not cast argument type everywhere, instead assign it to a variable after the instance check and use that variable.

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/pull/12441/commits/226c49a261240dc5305f96d69aadb6f86edfae84